qiskit/circuit/__init__.py

```diff
@@ -10,38 +10,62 @@
 # copyright notice, and modified files need to carry a notice indicating
 # that they have been altered from the originals.
 
-"""
-========================================
-Quantum Circuits (:mod:`qiskit.circuit`)
-========================================
+r"""
+=============================================
+Quantum circuit model (:mod:`qiskit.circuit`)
+=============================================
 
 .. currentmodule:: qiskit.circuit
 
-Overview
-========
+The fundamental element of quantum computing is the *quantum circuit*.  This is a computational
+routine that can be run, one shot at a time, on a quantum processing unit (QPU).  A circuit will act
+on a predefined amount of quantum data (in Qiskit, we only directly support qubits) with unitary
+operations (gates), measurements and resets.  In addition, a quantum circuit can contain operations
+on classical data, including near-time computations and control-flow constructs, which are executed
+by the controllers of the QPU.
+
+.. note::
+
+    You may wish to skip the introductory material and jump directly to:
+
+    * :ref:`the API overview of the whole circuit module <circuit-module-api>`
+    * :ref:`the detailed discussion about how circuits are represented <circuit-repr>`
+    * the core :class:`QuantumCircuit` class for how to build and query circuits
+    * :ref:`information on construction custom instructions <circuit-custom-gates>`
+    * :ref:`ways to work with circuit-level objects <circuit-working-with>`
+    * :ref:`discussion of Qiskit conventions for circuits, matrices and state labelling
+      <circuit-conventions>`
+
+Circuits are the low-level construct that are used to build up the higher level :ref:`primitives of
+quantum computation <qiskit-primitives>`, which accumulate many shots of quantum-circuit execution
+along with advanced error-mitigation techniques and measurement optimizations.
+
+In Qiskit, circuits can be defined in one of two regimes:
+
+* an *abstract* circuit, which is defined in terms of *virtual qubits* and arbitrary high-level
+  operations, like encapsulated algorithms and user-defined gates.
+
+* a *physical* circuit, which is defined in terms of the *hardware qubits* of one particular
+  backend, and contains only operations that this backend natively supports.  You might also see
+  this concept referred to as an *ISA circuit*.
```

Comment:
```suggestion
In Qiskit, quantum circuits are often distinguished by their compatibility with the QPU:

* an *abstract* quantum circuit, which is defined in terms of *virtual qubits* and arbitrary high-level
  operations, like encapsulated algorithms and user-defined gates.

* a *physical* quantum circuit, which is defined in terms of the *hardware qubits* of one particular
  backend, and contains only operations that this backend natively supports.  You might also see
  this concept referred to as an *ISA circuit*, i.e. a quantum circuit hat is compatible to a certain QPU.
```

Reply:
I don't think we need the additional text on the end here - it's just repeating the previous sentence. I'm worried that the change in the first sentence also weakens the distinction, which is pretty vital for us to make clear with the changes to submission that are coming in Runtime.

Reply:
I tend to disagree that it is only repeating the former sentence. This is the third paragraph of the text and there are two terms that users may not be familiar with ("hardware qubits" and "ISA"). I therefore thought it makes sense to add one sentence for uninitiated users.

I see your second point but was wondering what "regime" might be in "can be defined in one of two regimes". However it can be inferred by the enumeration so this is just a minor point in my opinion...

Reply:
I guess I'd rather leave additional definitions for the full definitions section below.  Perhaps an alternative is to move this little lead-in to below the definitions, if you think it's important that everything is exactly defined before use?

Reply:
I think it is fine. :-)